cmd/k8s-operator/proxygroup.go

```diff
@@ -470,20 +473,45 @@ func (r *ProxyGroupReconciler) ensureConfigSecretsCreated(ctx context.Context, p
 			return "", err
 		}
 
-		var authKey string
+		var authKey *string
 		if existingCfgSecret == nil {
 			logger.Debugf("Creating authkey for new ProxyGroup proxy")
 			tags := pg.Spec.Tags.Stringify()
 			if len(tags) == 0 {
 				tags = r.defaultTags
 			}
-			authKey, err = newAuthKey(ctx, r.tsClient, tags)
+			key, err := newAuthKey(ctx, r.tsClient, tags)
 			if err != nil {
 				return "", err
 			}
+			authKey = &key
+		}
+
+		// Get state Secret to check if it's already authed.
+		stateSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pgStateSecretName(pg.Name, i),
+				Namespace: r.tsNamespace,
+			},
+		}
+		if err := r.Get(ctx, client.ObjectKeyFromObject(stateSecret), stateSecret); err != nil && !apierrors.IsNotFound(err) {
+			return "", err
+		}
+		if authKey == nil && shouldRetainAuthKey(stateSecret) && existingCfgSecret != nil {
+			authKey, err = authKeyFromSecret(existingCfgSecret)
+			if err != nil {
+				return "", fmt.Errorf("error retrieving auth key from existing config Secret: %w", err)
+			}
+		}
+
+		// AdvertiseServices config is set by ingress-pg-reconciler, so make sure we
+		// don't overwrite it if already set.
+		existingAdvertiseServices, err := extractAdvertiseServicesConfig(existingCfgSecret, 106)
```

Comment:
Could we make `106` a const in this file? I feel like we'll likely forget to bump it here